core/src/main/java/org/apache/iceberg/FileScanTaskParser.java

```diff
@@ -21,116 +21,84 @@
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.databind.JsonNode;
 import java.io.IOException;
-import org.apache.iceberg.expressions.Expression;
-import org.apache.iceberg.expressions.ExpressionParser;
-import org.apache.iceberg.expressions.Expressions;
-import org.apache.iceberg.expressions.ResidualEvaluator;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
-import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.base.Strings;
 import org.apache.iceberg.util.JsonUtil;
 
 public class FileScanTaskParser {
-  private static final String SCHEMA = "schema";
-  private static final String SPEC = "spec";
-  private static final String DATA_FILE = "data-file";
-  private static final String START = "start";
-  private static final String LENGTH = "length";
-  private static final String DELETE_FILES = "delete-files";
-  private static final String RESIDUAL = "residual-filter";
+  private static final String TASK_TYPE = "task-type";
+
+  private enum TaskType {
```

Comment:
I feel the Java class names are starting to get confusing, even at the interface level. Although `DataTask` extends `FileScanTask`, there is really no file that the `DataTask` is scanning. So although `StaticDataTask` and `BaseFileScanTask` both implements `FileScanTask`, they are not really conceptually similar. The actual relationship is more like `StaticDataTask` implements `DataTask`, `BaseFileScanTask` implements `FileScanTask`, but `DataTask` and `FileScanTask` have many things in common so we made  `DataTask` extends `FileScanTask`.

To resolve this confusing situation and potentially accommodate other types of scan task, I am thinking if we should go one more layer above, so we keep the file scan task spec as is, and on top of that, have a serialization spec for **scan task**, which has types like `file-scan-task`, `data-task`. And we then have `ScanTaskParser` that delegates to the existing `FileScanTaskParser` or the `DataTaskParser`. What do you think?

Reply:
@jackye1995 I agree that `ScanTaskParser` is more accurate here. The only problem is that `FileScanTaskParser` is a public class in core module. Technically, we are breaking compatibility.

Reply:
actually, we can just deprecate the public methods from `FileScanTaskParser` and still keep the file scan task impl here.

Reply:
class name refactoring is done.